src/kernel/building.test.c

```diff
@@ -382,16 +385,59 @@ static void test_btype_defaults(CuTest *tc) {
     test_cleanup();
 }
 
+static void test_building_type_exists(CuTest * tc)
+{
+    region *r;
+    building *b;
+    building_type *btype, *btype2;
+
+    test_cleanup();
+    test_create_world();
+
+    btype2 = bt_get_or_create("lighthouse");
+    btype = bt_get_or_create("castle");
+
+    r = findregion(-1, 0);
+    b = new_building(btype, r, default_locale);
+
+    CuAssertPtrNotNull(tc, b);
+    CuAssertTrue(tc, !buildingtype_exists(r, NULL, false));
+    CuAssertTrue(tc, buildingtype_exists(r, btype, false));
+    CuAssertTrue(tc, !buildingtype_exists(r, btype2, false));
```

Comment:
Der dritte Parameter der Funktion (active) wird in diesem Test nicht getestet, und angesichts der relativ komplexen Bedingung in `buildingtype_exists` wäre das vielleicht nicht verkehrt.